src/core/exception.d

```diff
@@ -649,3 +643,24 @@ extern (C)
         onSwitchError(m.name, line);
     }
 }
+
+// TLS storage shared for all errors, chaining might create circular reference
+private void[128] _store;
+
+// only Errors for now as those are rarely chained
+private T staticError(T, Args...)(auto ref Args args)
+    if (is(T : Error))
+{
+    // pure hack, what we actually need is @noreturn and allow to call that in pure functions
+    static T get()
+    {
+        static assert(__traits(classInstanceSize, T) <= _store.length,
+                      T.stringof ~ " is too large for staticError()");
+
+        _store[0 .. __traits(classInstanceSize, T)] = typeid(T).init[];
+        return cast(T) _store.ptr;
```

Comment:
It doesn't seem to be a problem, the array is 16-byte aligned and we don't align classes ([see](https://github.com/D-Programming-Language/druntime/blob/dbd72a2f51d7b70449367947ca556aebdd3ee063/src/rt/lifetime.d#L71)).